chainer/testing/unary_math_function_test.py

```diff
@@ -0,0 +1,184 @@
+import numpy
+import unittest
+
+import chainer
+from chainer import cuda
+from chainer.testing import attr
+from chainer.testing import condition
+
+
+def make_data_default(shape, dtype):
+    x = numpy.random.uniform(-1, 1, shape).astype(dtype)
+    gy = numpy.random.uniform(-1, 1, shape).astype(dtype)
+    return x, gy
+
+
+def unary_math_function_test(func, func_expected=None, label_expected=None,
+                             make_data=None):
+    """Decorator for testing unary mathematical Chainer functions.
+
+    This decorator makes test classes test unary mathematical Chainer
+    functions. Tested are forward and backward computations on CPU and GPU
+    across parameterized ``shape`` and ``dtype``.
+
+    Args:
+        func(~chainer.Function): Chainer function to be tested by the decorated
+            test class.
+        func_expected: Function used to provide expected values for
+            testing forward computation. If not given, a corresponsing numpy
+            function for ``func`` is implicitly picked up by its class name.
+        label_expected(string): String used to test labels of Chainer
+            functions. If not given, the class name of ``func`` lowered is
+            implicitly used.
+        make_data: Function to customize input and gradient data used
+            in the tests. It takes ``shape`` and ``dtype`` as its arguments,
+            and returns a tuple of input and gradient data. By default, uniform
+            destribution ranged ``[-1, 1]`` is used for both.
+
+    The decorated test class tests forward and backward computations on CPU and
+    GPU across the following :func:`~chainer.testing.parameterize` ed
+    parameters:
+
+    - shape: rank of zero, and rank of more than zero
+    - dtype: ``numpy.float16``, ``numpy.float32`` and ``numpy.float64``
```

Comment:
Yes, it would be. As discussed, it is supposed to be enough to support testing integer types when it would be actually needed.